proposals/lip_define_state_and_state_transitions_of_dpos_module.md

```diff
@@ -898,52 +898,351 @@ shuffleValidatorsList(validatorsAddresses, randomSeed):
+                            "properties": {
+                                "delegateAddress": {
+                                    "dataType": "bytes",
+                                    "fieldNumber": 1
+                                },
+                                "delegateWeight": {
+                                    "dataType": "uint64",
+                                    "fieldNumber": 2
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "genesisData": {
+            "type": "object",
+            "fieldNumber": 4,
+            "required": ["initRounds", "initDelegates"],
+            "properties": {
+                "initRounds": {
+                    "dataType": "uint32",
+                    "fieldNumber": 1
+                },
+                "initDelegates": {
+                    "type": "array",
+                    "fieldNumber": 2,
+                    "items": { "dataType": "bytes" }
+                }
+            }
+        }
+    }
+}
+```
 
 
-#### After Genesis Block Execution
+#### Genesis State Initialization
+
+During the genesis state initialization stage, the following steps are executed. If any step fails, the block is discarded and has no further effect.
+
+Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets for the DPoS module and let `genesisBlockAssetObject` be the deserialization of `genesisBlockAssetBytes` according to the `genesisDPoSStoreSchema` schema, given above.
+
+* Initial checks on the properties of `genesisBlockAssetObject`:
+    * Across elements of the `validators` array, all `address` values must be unique, all `name` values must also be unique.
+    * For all elements of the `validators` array:
+        * `address` values must have length `ADDRESS_LENGTH`.
+        * `name` values must have length between `1` and `MAX_LENGTH_NAME` (included), and must contain only characters from the set `abcdefghijklmnopqrstuvwxyz0123456789!@$&_.`.
+        * `generatorKey` values must have length 32.
+        * `blsKey` values must have length 48.
+        * `proofOfPossession` values must have length 96.
+    * Across elements of the `voters` array, all `address` values must be unique, and must have length `ADDRESS_LENGTH`.
+    * For all elements of the `voters` array:
+        * Across elements of the `sentVotes` array, all `delegateAddress` values must be unique, and must have length `ADDRESS_LENGTH`. 
+        * `sentVotes` has size is at most `MAX_NUMBER_SENT_VOTES`.
+        * `sentVotes` must be in lexicographic order of `delegateAddress`.
+        * `pendingUnlocks` has size is at most `MAX_NUMBER_PENDING_UNLOCKS`.
+        * `pendingUnlocks` must be ordered by lexicographical order of `delegateAddress`, ties broken by increasing `amount`, ties broken by increasing `unvoteHeight`. 
```

Comment:
I think we should indicate that these operations must be in order. This way it doesn't tell what tie is broken after what tie.

Reply:
ok, I added a few words to the sentence to indicate the order of tie breaking, tell me if it's fine for you.